tornado-drivers/opencl/src/main/java/uk/ac/manchester/tornado/drivers/opencl/scheduler/OCLScheduler.java

```diff
@@ -34,10 +35,16 @@ public class OCLScheduler {
     private static final String NVIDIA = "NVIDIA";
 
     private static OCLKernelScheduler getInstanceGPUScheduler(final OCLDeviceContext context) {
-        if (context.getDevice().getDeviceVendor().contains(AMD_VENDOR)) {
+        OCLTargetDevice device = context.getDevice();
+        if (device.getDeviceVendor().contains(AMD_VENDOR)) {
             return new OCLAMDScheduler(context);
-        } else if (context.getDevice().getDeviceVendor().contains(NVIDIA)) {
-            return new OCLNVIDIAGPUScheduler(context);
+        } else if (device.getDeviceVendor().contains(NVIDIA)) {
+            int majorVersion = Integer.parseInt(device.getDriverVersion().split("\\.")[0]);
+            if (majorVersion >= 550) {
```

Comment:
Can we make this 550 a global configuration or move to TornadoOptions instead of hardcoded?